### ISSUE TITLE
Feature request: set default grid comp reference time to return alarm true on first timestep

### ISSUE THREAD

**Issue by lizziel** [GEOS-ESM/MAPL#279]
This feature request follows up on an issue I was having with MAPL alarms in GCHP.

**Background:**
In the GIGCchem grid comp within GCHP the GIGCchem_DT is always set to twice the heartbeat. Every time the grid comp run method is called the MAPL alarm is queried with:

```
CALL MAPL_Get(STATE, RUNALARM=ALARM, __RC__)
IsChemTime = ESMF_AlarmIsRinging(ALARM, __RC__)
```

The expected behavior is true in the first call to run, false in the second, true in the third, and so on, alternating between true and false. My original issue was that halving GIGCchem_DT switched the true/false pattern. I traced the issue to GIGCchem_REFERENCE_TIME set in GCHP.rc.
 
**Observations:**
I figured out that the MMSS of GIGCchem_REFERENCE_TIME must match the GIGCchem_DT timestep to get correct behavior in MAPL. To reiterate, the correct behavior is if GIGCchem_DT is twice HEARTBEAT_DT then GIGCchem alarm is true for run call numbers 0, 2, 4, etc, and false otherwise.
 
For example, for GIGCchem_DT = 1200 and HEARTBEAT_DT = 600, GIGCchem_REFERENCE_TIME must be 121000. Using the default value of 120000 gives the wrong behavior, with true for run call numbers 1, 3, 5, etc, skipping the first timestep.
 
Another example (and the one I originally had issue with): For GIGCchem_DT = 600 and HEARTBEAT_DT = 300, GIGCchem_REFERENCE_TIME must be 120500. Using 120000 or 121000 result in the wrong behavior, giving alarm true for run call numbers 1, 3, 5, etc.
 
This has not been an issue for GEOS-Chem in GEOS since GIGCchem_DT is the same as the heartbeat in that model. For GCHP, however, we always use a GIGCchem_DT that is double the heartbeat, which leads to these reference time complications.
 
**Feature request:**
Would you be open to changing the default so that in the absence of a reference time each grid comp alarm is always true in the first timestep? Following that it would be true or false based on its frequency relative to the heartbeat. For example:
 
Heartbeat = x
GridComp1_DT = nx
GridComp1 alarm is true for run calls 0, n, 2n, etc
 
In other words, for a 5 minute heartbeat and a grid comp with 15 min frequency, the grid comp alarm would be true at 00:00, 15:00, 30:00,etc.

**Comment by atrayano** [comment]
@lizziel, 
1) You correctly described the current behavior of the alarms in MAPL. Do accomplish the behavior you want, the REFERENCE_TIME has to be set to the START_TIME of your experiment + the HEARTBEAT of the clock

2) Unfortunately, if we change the default behavior of alarms in MAPL, this will affect GEOS in many way and I do not think it is a good idea to make such global change
3) There are several possibilities to solve this issue:
- we can create an alarm inside the particular component (not a MAPL alarm)
- we can change the resource file, either by script or manually, to set the REFERENCE_TIME accordingly
- we can add code inside your gridded component to overwrite the REFERENCE_TIME for the specific component, as long as this is done prior to the call to MAPL_GenericInitialize. I have done similar task in GOCART (the value there is different!)

I will consult with @tclune before we make a final decision

**Comment by lizziel** [comment]
I found that if I set the reference time to HHMMSS of the heartbeat then I get the desired result. There is no need to add the start time of the experiment to the heartbeat. Since I already run a pre-processing script to set HHMMSS of the heartbeat to the GEOS-Chem dynamic timestep, it is a simple step to also use the same value for reference time. Thanks for your help!
